packages/url_launcher/url_launcher_web/test/url_launcher_web_test.dart

```diff
@@ -146,6 +145,20 @@ void main() {
         verify(mockWindow.open('sms:+19725551212?body=hello%20there', ''));
       });
 
+      test('setting webOnlyLinkTarget as _self opens the url in the same tab',
+          () {
+        plugin.openNewWindow("https://www.google.com",
+            webOnlyLinkTarget: "_self");
+        verify(mockWindow.open('https://www.google.com', '_self'));
+      });
+
+      test('setting webOnlyLinkTarget as _blank opens the url in a new tab',
```

Comment:
This is exactly what needs to be tested, thanks!

PS: Once you change the logic above (use whatever the user passed instead of the automatic detection), can you add a new test in the `Safari` group below to see that if the user passes a '_blank' to a `mailto:` url, the plugin will use '_blank' instead of '_top'?